SPM Example/PortalSwift/ViewController.swift

```diff
@@ -220,29 +230,151 @@ class ViewController: UIViewController, UITextFieldDelegate {
-    ) else {
-      throw URLError(.badURL)
+    let backupWithPortal = client.environment?.backupWithPortalEnabled ?? false
+
+    if !backupWithPortal {
+      guard let cipherTextUrl = URL(
+        string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/cipher-text/fetch?backupMethod=\(withBackupMethod.rawValue)"
+      ) else {
+        throw URLError(.badURL)
+      }
+      let cipherTextData = try await requests.get(cipherTextUrl)
+      let cipherTextResponse = try decoder.decode(CipherTextResult.self, from: cipherTextData)
+
+      cipherText = cipherTextResponse.cipherText
+
+      guard let organizationBackupShareUrl = URL(
+        string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/org-share/fetch?backupMethod=\(withBackupMethod.rawValue)-SECP256K1"
+      ) else {
+        throw URLError(.badURL)
+      }
+      let organizationBackupShareData = try await requests.get(organizationBackupShareUrl)
+      let organizationBackupShareResponse = try decoder.decode(OrgShareResult.self, from: organizationBackupShareData)
+
+      organizationShare = organizationBackupShareResponse.orgShare
+    } else {
+      var walletId: String? = nil
+
+      for wallet in client.wallets {
+        if wallet.curve == .SECP256K1 {
+          for backupSharePair in wallet.backupSharePairs {
+            if backupSharePair.status == .completed, backupSharePair.backupMethod == withBackupMethod {
+              walletId = wallet.id
+            }
+          }
+        }
+      }
+
+      guard let walletId = walletId else {
+        throw PortalExampleAppError.clientInformationUnavailable("Could not find backup share for backup method.")
+      }
+
+      guard let prepareEjectUrl = URL(string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/prepare-eject") else {
+        throw URLError(.badURL)
+      }
+      let prepareEjectData = try await requests.post(prepareEjectUrl, withBearerToken: nil, andPayload: ["walletId": walletId])
+      guard let prepareEjectResponse = String(data: prepareEjectData, encoding: .utf8) else {
+        throw PortalExampleAppError.couldNotParseCustodianResponse("Unable to read prepare eject response.")
+      }
+
+      print("Wallet ejectable until \(prepareEjectResponse)")
     }
-    let organizationBackupShareData = try await requests.get(organizationBackupShareUrl)
-    let organizationBackupShareResponse = try decoder.decode(OrgShareResult.self, from: organizationBackupShareData)
 
     let privateKey = try await portal.eject(
       withBackupMethod,
-      withCipherText: cipherTextResponse.cipherText,
-      andOrganizationBackupShare: organizationBackupShareResponse.orgShare
+      withCipherText: cipherText,
+      andOrganizationBackupShare: organizationShare
+    )
+
+    return privateKey
+  }
+
+  public func ejectSolana(_ withBackupMethod: BackupMethods) async throws -> String {
```

Comment:
oooh, just realized that eject solana is separate, Any reason for that?
As per my understanding, we need one eject function to eject both `ejectPrivateKeys()`
Is it doable to eject ETH only and keep Solana?